index.d.ts

```diff
@@ -132,32 +132,22 @@ declare namespace html {
    * html = <Unclean repeat="a" n={5} />
    * ```
    *
-   * @param {(proxy: any) => string} htmlFn
+   * @param {Function} htmlComponent the *clean* component to compile.
    * @param {boolean} [strict=true] if we should throw an error when a property is not found.
    * @param {string | undefined} [separator] the string used to interpolate and separate parameters
-   * @returns {function} the compiled template function
+   * @returns {Function} the compiled template function
    * @this {void}
    */
-  export function compile<
-    P extends
-      | Record<string, any>
-      | string[]
-      | ((args: Record<string, string>) => JSX.Element) = {}
-  >(
+  export function compile<P extends StringPropsOnly<P>>(
```

Comment:
I think we could inline this here

```suggestion
  export function compile<P extends { [K in keyof P]: string | number | boolean }>(
```

Reply:
I think I'm going back with allowing non string types, it could give the false perception that the real props are going there...